crates/ark/src/data_explorer/r_data_explorer.rs

```diff
@@ -336,28 +333,37 @@ impl RDataExplorer {
                 Rf_findVarInFrame(env, sym)
             };
 
-            let old = self.table.get().sexp;
-            if new == old {
-                None
+            let old = self.table.get();
+            let old = unwrap!(old, Err(_) => {
+                // This is AFAICT impossible because the table is only deleted when the data explorer instance is
+                // deleted and this method belongs to that data explorer instance.
+                log::error!("Old table has been deleted? This is unexpected, but we'll update the data explorer table.");
+                // It's `unsafe` because we RObject::new calls protect, and it shouldn't
+                // be called outside of the R main thread.
+                self.table.set(unsafe { RObject::new(new) });
```

Comment:
Can we change `set()` to take a thread safe object so it's clear what's going on?